pinns/pinns.c

```diff
@@ -126,11 +143,31 @@ int main(int argc, char **argv) {
       return EXIT_FAILURE;
     }
   }
+
   if (bind_user) {
     if (bind_ns(pin_path, "user") < 0) {
       return EXIT_FAILURE;
     }
   }
 
+  if (bind_pid) {
+    if (fork() < 0) {
+      pexit("unable to fork process");
+    }
+    // The first child created by a process after a call to unshare(2) using
+    // the CLONE_NEWPID flag) has the PID 1, and is the init process for the
+    // namespace. We just have to pin the process which is not PID 1 after the
+    // call to fork(2).
+    if (getpid() != 1 && bind_ns(pin_path, "pid") < 0) {
```

Comment:
I think it should be the child to do the pinning.

You can use directly the return code from fork().

```
pid_t p = fork();
....

if (p == 0 && bind_ns(pin_path, "pid") < 0) {
```

Reply:
Cool thank you for the hint. Unfortunately the pid namespace unsharing does not seem to be a valid use case any more.